src/zepben/evolve/streaming/get/async_utils.py

```diff
@@ -0,0 +1,30 @@
+#  Copyright 2022 Zeppelin Bend Pty Ltd
+#
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+import asyncio
+import functools
+from concurrent.futures import ThreadPoolExecutor
+from typing import Callable, TypeVar, Awaitable
+
+__all__ = ["make_async"]
+
+
+T = TypeVar('T')
+
+
+def make_async(func: Callable[..., T], timeout=None) -> Callable[..., Awaitable[T]]:
```

Comment:
This breaks pretty horribly. I don't know why it doesn't break in tests - something we need to look into.

```
Traceback (most recent call last):
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi", line 182, in grpc._cython.cygrpc.get_working_loop
RuntimeError: no running event loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/krut/work/git/cim-to-opendss-processor/src/zepben/cim2opendss/main.py", line 324, in <module>
    start()
  File "/home/krut/work/git/cim-to-opendss-processor/src/zepben/cim2opendss/main.py", line 320, in start
    asyncio.run(main())
  File "/usr/lib64/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib64/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/home/krut/work/git/cim-to-opendss-processor/src/zepben/cim2opendss/main.py", line 272, in main
    network_hierarchy = (await client.get_network_hierarchy()).throw_on_error().value
  File "/home/krut/work/git/cimbend/src/zepben/evolve/streaming/grpc/grpc.py", line 78, in throw_on_error
    raise self.result
  File "/home/krut/work/git/cimbend/src/zepben/evolve/streaming/grpc/grpc.py", line 106, in try_rpc
    return GrpcResult(await rpc())
  File "/home/krut/work/git/cimbend/src/zepben/evolve/streaming/get/network_consumer.py", line 413, in _handle_network_hierarchy
    response = await make_async(self._stub.getNetworkHierarchy)(GetNetworkHierarchyRequest())
  File "/home/krut/work/git/cimbend/src/zepben/evolve/streaming/get/async_utils.py", line 28, in inner
    return await asyncio.wait_for(loop.run_in_executor(executor, functools.partial(func, *args, **kwargs)), timeout=timeout)
  File "/usr/lib64/python3.9/asyncio/tasks.py", line 442, in wait_for
    return await fut
  File "/usr/lib64/python3.9/concurrent/futures/thread.py", line 52, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/krut/work/git/cim-to-opendss-processor/venv/lib64/python3.9/site-packages/grpc/aio/_channel.py", line 132, in __call__
    call = UnaryUnaryCall(request, _timeout_to_deadline(timeout),
  File "/home/krut/work/git/cim-to-opendss-processor/venv/lib64/python3.9/site-packages/grpc/aio/_call.py", line 503, in __init__
    channel.call(method, deadline, credentials, wait_for_ready),
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/channel.pyx.pxi", line 134, in grpc._cython.cygrpc.AioChannel.call
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi", line 49, in grpc._cython.cygrpc._AioCall.__cinit__
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi", line 97, in grpc._cython.cygrpc.init_grpc_aio
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi", line 101, in grpc._cython.cygrpc.init_grpc_aio
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/grpc_aio.pyx.pxi", line 86, in grpc._cython.cygrpc._initialize_per_loop
  File "src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi", line 184, in grpc._cython.cygrpc.get_working_loop
  File "/usr/lib64/python3.9/asyncio/events.py", line 642, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'ThreadPoolExecutor-0_0'.
```

Reply:
No longer an issue as this has been removed